src/plugins/discover/public/application/components/doc_viewer_links/doc_viewer_links.tsx

```diff
@@ -13,7 +13,11 @@ export function DocViewerLinks(renderProps: DocViewLinkRenderProps) {
   const listItems = getDocViewsLinksRegistry()
     .getDocViewsLinksSorted()
     .map((item) => {
-      item.href = item.generateurlfn ? item.generateurlfn(renderProps) : item.href;
+      if (item.generateurlcb) {
+        item.href = item.generateurlcb(renderProps);
+        delete item.generateurlcb;
+      }
```

Comment:
Sorry i didnt catch this earlier but we should [avoid mutating](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#avoid-mutability-and-state) the docViewLink object directly. We can return a new object instead e.g.

```ts
.map((item) => {
  const { generateurlcb, href, ...props } = item;
  const listItem: EuiListGroupItemProps = {
    ...props,
    href: generateurlcb ? generateurlcb(renderProps) : href,
  }

  return listItem;
}
```

This way we also ensure that the new object has only the props that the `EuiListGroup` component accepts.

Reply:
amazing, thanks

Reply:
amazing, thanks